src/components/Message/hooks/useHandleAction.js

```diff
@@ -0,0 +1,38 @@
+// @ts-check
+import { useContext } from 'react';
+import { ChannelContext } from '../../../context';
+
+export const handleActionWarning = `Action handler was called, but it is missing one of its required arguments.
+      Make sure the ChannelContext was properly set and that this hook was initialized with a valid message.`;
+/**
+ * @type {(message: import('stream-chat').MessageResponse) => (name: string, value: string, event: React.MouseEvent<HTMLElement>) => Promise<void>}
+ */
+export const useHandleAction = (message) => {
+  /**
+   *@type {import('types').ChannelContextValue}
+   */
+  const { channel, updateMessage, removeMessage } = useContext(ChannelContext);
+  return async (name, value, event) => {
+    event.preventDefault();
+    /**
+     *@type {import('types').ChannelContextValue}
+     */
+    if (!message || !updateMessage || !removeMessage || !channel) {
+      console.log(!!message, !!updateMessage, !!removeMessage, !!channel);
```

Comment:
Do we need this log statement? Its just bunch of true/false, doesn't say much!

Reply:
Some debugging leftover x)